src/slc/mailrouter/tests/test_injection.py

```diff
@@ -45,7 +44,9 @@ def do_inject(self, encoding=None):
         mailfile = open_mailfile("mail_plain.txt")
         mail = mailfile.read()
         mailfile.close()
-        mail = safe_encode(mail, encoding or "utf-8")
+        if six.PY2:
+            mail = mail.decode("utf-8")
```

Comment:
This will fail, you cannot write unicode in to a BytesIO:
```
[ale@emily ~]$ python2.7
Python 2.7.17rc1 (default, Oct 10 2019, 10:26:01) 
[GCC 9.2.1 20191008] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> from io import BytesIO
>>> BytesIO(u"a")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'unicode' does not have the buffer interface
```

Reply:
I'm not writing it to the BytesIO like that. Look at the next line, there it's encoded again. That's to make sure we get the correct encoding that's passed to the method.

Reply:
Right! I missed it